libobs/obs.c

```diff
@@ -2019,6 +2019,26 @@ obs_source_t *obs_get_source_by_name(const char *name)
 				   obs_source_addref_safe_);
 }
 
+obs_source_t *obs_get_source_by_uuid(const char *uuid)
+{
+	struct obs_source **first = &obs->data.first_source;
+	struct obs_source *source;
+
+	pthread_mutex_lock(&obs->data.sources_mutex);
+
+	source = *first;
+	while (source) {
+		if (strcmp(source->context.uuid, uuid) == 0) {
+			source = obs_source_addref_safe_(source);
```

Comment:
I have the feeling that `obs_source_addref_safe_` was only intended to be used with `get_context_by_name` calls. I realise this was probably copied from transitions but `obs_source_get_ref` seems like it might be better here.

Reply:
Yeah it probably just exists since the context functions need a callback that takes `void *`, I just updated it to use `obs_source_get_ref()` directly now.